vendor/github.com/vmware/govmomi/toolbox/service.go

```diff
@@ -110,14 +115,37 @@ func (s *Service) backoff() {
 	}
 }
 
-// Start initializes the RPC channels and starts a goroutine to listen for incoming RPC requests
-func (s *Service) Start() error {
+func (s *Service) stopChannel() {
+	_ = s.in.Stop()
+	_ = s.out.Stop()
+}
+
+func (s *Service) startChannel() error {
 	err := s.in.Start()
 	if err != nil {
 		return err
 	}
 
-	err = s.out.Start()
+	return s.out.Start()
```

Comment:
is it possible to have s.in.Start() succeeds but s.out.Start failed? And if that happens, do you need to stop the s.in first before return error?

Reply:
I don't know of a case where `out.Start` would error, other than if not running inside a VM.  But in that case, `in.Start` would have error'd first.

The `in` channel is for **vmx to guest** RPC, there can only be 1.  So `in.Start` can fail if another instance of toolbox (or vmtoolsd) is already registered.

The `out` channel is for the **guest to vmx** RPC, there is no limit that I'm aware of.

Reply:
I see the point. thanks